### DISCUSSION TITLE
Datenaustausch zwischen Solectrus und HomeAssistant

### DISCUSSION THREAD

**Discussion by Zeppelin-PV** [solectrus/solectrus#3717] [Q&A]
Ist, oder wäre es auch denkbar bzw. machbar, wenn einzelne der "Klein"-Verbraucher schon in HomeAssistant vorhanden sind, diese auf den Solectrus PI zu bringen ???
Das wäre natürlich eine grandiose Sache !!!

_Originally posted by @Zeppelin-PV in https://github.com/solectrus/solectrus/issues/3396#issuecomment-2501448919_

**Comment by Zeppelin-PV** [comment]
habe es nicht geschafft dies direkt in Diskussion zu verschieben

**Reply by ledermann** [reply]
Habe das Issue jetzt verschoben. Meine Antwort in #3396 war: 

> Die Messwerte für SOLECTRUS müssen zwingend in der eigenen InfluxDB stehen. Wie sie dahin kommen, ist egal. Wenn du also Werte in HomeAssistant hast, könntest du sie dort per MQTT bereitstellen und dann über den MQTT-Collector abholen.
>
> Wenn es sich um Messwerte eines Shellys handelt, kannst du sie aber auch über den Shelly-Collector abholen, selbst wenn dieser bereits vom HomeAssistant abgefragt wird (oder umgekehrt der Shelly selbst per MQTT die Messwerte dorthin schickt).
>
> Empfehlen würde ich den zweiten Weg, weil er einfacher ist und unabhängig von HomeAsistant funktioniert.

Ist denn noch etwas unklar?

**Reply by Zeppelin-PV** [reply]
Ja und Nein

Wenn die Option Kleinverbraucher kommt, habe ich schon zwei Verbraucher über einen Shelly in Betrieb, die auch schon in die influxdb geschrieben werden. Diese sind mir auch wichtig zu sehen. (Aufzug und Gastherme)

Für eventuell zwei weitere habe ich gedacht, dass diese irgendwie bei HomeAssistant abzuholen sind da sie dort schon in Betrieb sind. Es sind Strommessadapter (Zwischenstecker) von NOUS die im zigbee2mqtt Netzwerk auf HA laufen.
Sollte jemand dieselbe Idee verfolgen, und wissen wie das geht, wäre ich für jeden Tipp dankbar wie das zu bewerkstelligen ist.
Wenn dies doch zu kompliziert ist, müsste ich dann wohl oder übel die Zigbee Adapter durch Shelly's ersetzen.
Bei manchen Verbrauchern ist dies aber u.U. ein Problem, das die Shelly Zwischenstecker max. 2500W messen, der NOUS Zigbee Adapter dagegen 3650W

**Reply by ledermann** [reply]
Ok, in diesem Fall könntest du versuchen, die Messwerte aus dem HomeAssistant per MQTT verfügbar zu machen. In HomeAssistant kann **Mosquitto broker** als Add-on installiert werden. Wie das im Einzeln geht, müsstest du aber selbst herausfinden, es gibt viele Tutorials zum Thema. 

Wenn das MQTT-Setup dann steht, kannst du anschließend mit dem [MQTT-Collector](https://docs.solectrus.de/referenz/mqtt-collector/) die Messwerte abholen und zu SOLECTRUS bringen.

**Reply by Zeppelin-PV** [reply]
ich habe das schon so wie von Dir beschrieben am laufen
Beispiel Gefrierschrank
Strommessadapter NOUS 1AZ (Zigbee) an HomeAsisstant
Wird dort auch angezeigt
![grafik](https://github.com/user-attachments/assets/a5671f8e-c019-463d-bc0b-cfc79f4857aa)
![grafik](https://github.com/user-attachments/assets/a57bc528-2218-4d22-9f3f-c211fab143d3)
Über Zigbee2MQTT geht der Sensor auch an den Mosquitto Broker
![grafik](https://github.com/user-attachments/assets/deca5391-296a-4ea2-b490-23837461ebe6)
Das funktioniert auch soweit alles. Habe 19 solcher Messadapter im Haus verteilt.
Mein Problem: wie rufe ich die Daten vom HomeAssistant 192.168.1.90 ab.
da gibt es wohl Ports in dem Broker 1893, 1894, oder 8893, 8894 aber aufrufen kann ich da nichts.
Die Daten stehen irgendwo abholbereit, und ich habe keine Ahnung :(
Und dann müssten die Daten ja auf den PI4 mit Solectrus und dort in die Influx Datenbank geschrieben werden.

Das wäre einer meiner Weihnachtswünsche ;-)
so dass dies ab dem neuen Jahr in Solectrus integriert wäre

**Reply by ledermann** [reply]
Um ein Verständnis für MQTT zu bekommen, würde ich empfehlen, einen MQTT-Client (wie z.B. [MQTTX](https://mqttx.app/) oder [MQTT Explorer](https://mqtt-explorer.com/)) auszuprobieren und nachsehen, welche Topics bei deinem Broker zur Verfügung stehen. Der Zugriff auf MQTT (meist Port 1883) ist nicht über den Browser möglich, da braucht es spezielle Clients. 

Wenn du mit einem der genannten Clients Erfolg hast, ergibt sich auch recht schnell, wie der MQTT-Collector eingerichtet wird. Der ist auch nur ein MQTT-Client und sorgt dafür, dass die Werte permanent abgeholt und in InfluxDB gespeichert werden.

**Reply by Zeppelin-PV** [reply]
das ist ja ein tolles Tool, MQTT Explorer

zeigt mir folgendes an:
{
  "availability": [
    {
      "topic": "zigbee2mqtt/bridge/state",
      "value_template": "{{ value_json.state }}"
    }
  ],
  "device": {
    "identifiers": [
      "zigbee2mqtt_0xa4c13890ee46bb8f"
    ],
    "manufacturer": "Nous",
    "model": "Smart plug (with power monitoring) (A1Z)",
    "name": "Gefrierschrank",
    "via_device": "zigbee2mqtt_bridge_0x00124b002c44fbea"
  },
  "device_class": "power",
  "enabled_by_default": true,
  "entity_category": "diagnostic",
  "object_id": "gefrierschrank_power",
  "origin": {
    "name": "Zigbee2MQTT",
    "sw": "1.42.0",
    "url": "https://www.zigbee2mqtt.io"
  },
  "state_class": "measurement",
  "state_topic": "zigbee2mqtt/Gefrierschrank",
  "unique_id": "0xa4c13890ee46bb8f_power_zigbee2mqtt",
  "unit_of_measurement": "W",
  "value_template": "{{ value_json.power }}"

ist das womöglich der Pfad den ich brauche ?
![grafik](https://github.com/user-attachments/assets/a57b6f6a-5fe0-4dae-91af-1ce5472fb4a7)

**Reply by ledermann** [reply]
Du bist auf dem richtigen Weg. Was du konkret benötigst, sind die genauen Topics, unter denen die Messwerte bereitgestellt werden. Ein Topic ist eine Zeichenkette, üblicherweise als Pfadangabe mit einem oder mehreren "/" geschrieben. Möglicherweise ist `zigbee2mqtt/Gefrierschrank´ das, was du brauchst. Die Details musst du aber selber herausfinden. 

Mit diesen Informationen kannst du dann den MQTT-Collector einrichten: https://docs.solectrus.de/referenz/mqtt-collector/

**Reply by Zeppelin-PV** [reply]
sorry, ich komm da nicht weiter
unklar ist, bevor ich mir meine Instanz zerschieße, ob alle wie in der MQTT Beschreibung genannten Parameter gebraucht werden, wenn doch nur "power" abgeholt wird 
sähe es dann so aus ???
.env
MQTT_HOST=192.168.1.90
MQTT_PORT=1883
MQTT_USERNAME=xxxx
MQTT_PASSWORD=xxxx
INFLUX_HOST=influxdb
INFLUX_SCHEMA=http
INFLUX_PORT=8086
INFLUX_ADMIN_TOKEN=my-super-secret-admin-token ????

MAPPING_0_TOPIC=zigbee2mqtt/Gefrierschrank
MAPPING_0_MEASUREMENT=Gefrierschrank
MAPPING_0_FIELD=power
MAPPING_0_TYPE=float
und in der docker-compose.yml - was könnte da alles weggelassen werden ?
environment:
      - TZ
      - INFLUX_SCHEMA
      - INFLUX_HOST
      - INFLUX_TOKEN
      - INFLUX_ORG
      - INFLUX_BUCKET
      - INFLUX_PORT
      - MQTT_HOST
      - MQTT_PORT
      - MQTT_SSL
      - MQTT_USERNAME
      - MQTT_PASSWORD
      - MAPPING_0_TOPIC
      - MAPPING_0_JSON_KEY
      - MAPPING_0_JSON_PATH
      - MAPPING_0_JSON_FORMULA
      - MAPPING_0_MEASUREMENT
      - MAPPING_0_MEASUREMENT_POSITIVE
      - MAPPING_0_MEASUREMENT_NEGATIVE
      - MAPPING_0_FIELD
      - MAPPING_0_FIELD_POSITIVE
      - MAPPING_0_FIELD_NEGATIVE
      - MAPPING_0_TYPE
      - MAPPING_0_MIN
      - MAPPING_0_MAX

**Reply by ledermann** [reply]
Sieht alles richtig aus. In der `docker-compose.yml` müssen die Variablen aufgelistet werden, auf die der Container Zugriff haben soll. Du kannst also diejenigen weglassen, die es in der `.env` gar nicht gibt. Es schadet aber auch nicht, wenn zu viele enthalten sind.

**Reply by Zeppelin-PV** [reply]
müsste in der influx nicht irgendwie MQTT auftauchen ?
![grafik](https://github.com/user-attachments/assets/990f2015-5c0f-4ac3-9207-fbbec35c0133)

in Solectrus erscheint (natürlich) dies ...
![grafik](https://github.com/user-attachments/assets/e275b366-2d76-4742-ba7b-77de081b4caa)

.env
#################################################################
###             MQTT Collector                                ###
#################################################################
#
MQTT_INTERVAL=5
MQTT_HOST=192.168.1.90
MQTT_PORT=1883
MQTT_USERNAME=Ernst
MQTT_PASSWORD=15103662
INFLUX_HOST=influxdb
INFLUX_SCHEMA=http
INFLUX_PORT=8086
INFLUX_ADMIN_TOKEN=my-super-secret-admin-token
#
MAPPING_0_TOPIC=zigbee2mqtt/Gefrierschrank
MAPPING_0_MEASUREMENT=Gefrierschrank
MAPPING_0_FIELD=power
MAPPING_0_TYPE=float
#
compose.yml
  mqtt-collector:
    image: ghcr.io/solectrus/mqtt-collector:latest
    environment:
      - TZ
      - INFLUX_SCHEMA
      - INFLUX_HOST
      - INFLUX_TOKEN
      - INFLUX_ORG
      - INFLUX_BUCKET
      - INFLUX_PORT
      - MQTT_HOST
      - MQTT_PORT
      - MQTT_SSL
      - MQTT_USERNAME
      - MQTT_PASSWORD
      - MAPPING_0_TOPIC
      - MAPPING_0_JSON_KEY
      - MAPPING_0_JSON_PATH
      - MAPPING_0_JSON_FORMULA
      - MAPPING_0_MEASUREMENT
      - MAPPING_0_MEASUREMENT_POSITIVE
      - MAPPING_0_MEASUREMENT_NEGATIVE
      - MAPPING_0_FIELD
      - MAPPING_0_FIELD_POSITIVE
      - MAPPING_0_FIELD_NEGATIVE
      - MAPPING_0_TYPE
      - MAPPING_0_MIN
      - MAPPING_0_MAX
      - ... # weitere Mappings bei Bedarf
    logging:
      options:
        max-size: 10m
        max-file: '3'
    restart: unless-stopped
    depends_on:
      influxdb:
        condition: service_healthy
    links:
      - influxdb
    labels:
      - com.centurylinklabs.watchtower.scope=solectrus

mit welchem Log kann ich nachschauen ob da was rüber kommt, vom HomeAssistant ?

**Reply by ledermann** [reply]
Das Maß aller Dinge und allererste Anlaufstelle für Probleme jeder Art sind die Docker-Logs, in diesem Fall vom MQTT-Collector. Wenn der Collector irgendwas empfängt, taucht das im Log auf: `docker compose logs mqtt-collector`

Wenn der Collector nichts empfängt, stimmt etwas mit der Verbindung zum MQTT-Broker nicht oder das Topic existiert dort nicht.

**Reply by Zeppelin-PV** [reply]
diese Botschaft zeigt mir der mqtt-collector log
und das zigmal untereinander
irgendwas mit INFLUX_ORG und INFLUX_TOKEN
muss da noch die INFLUX_ORG:solectrus in die .env ???
und darf das nicht INFLUX_ADMIN_TOKEN=my-super-secret-admin-token   heißen ???

mqtt-collector-1  | MQTT collector for SOLECTRUS, Version 0.4.1, built at 2024-11-09T06:42:39.089Z
mqtt-collector-1  | https://github.com/solectrus/mqtt-collector
mqtt-collector-1  | Copyright (c) 2023-2024 Georg Ledermann and contributors, released under the MIT License
mqtt-collector-1  | 
mqtt-collector-1  | bundler: failed to load command: app.rb (app.rb)
mqtt-collector-1  | /mqtt-collector/lib/config.rb:59:in `fetch': key not found: "INFLUX_TOKEN" (KeyError)
mqtt-collector-1  | Did you mean?  "INFLUX_ORG"
mqtt-collector-1  | 	from /mqtt-collector/lib/config.rb:59:in `initialize'
mqtt-collector-1  | 	from app.rb:22:in `new'
mqtt-collector-1  | 	from app.rb:22:in `<top (required)>'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli/exec.rb:58:in `load'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli/exec.rb:58:in `kernel_load'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli/exec.rb:23:in `run'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli.rb:456:in `exec'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli.rb:35:in `dispatch'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/cli.rb:29:in `start'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/exe/bundle:28:in `block in <top (required)>'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
mqtt-collector-1  | 	from /usr/local/bundle/gems/bundler-2.5.23/exe/bundle:20:in `<top (required)>'
mqtt-collector-1  | 	from /usr/local/bundle/bin/bundle:25:in `load'
mqtt-collector-1  | 	from /usr/local/bundle/bin/bundle:25:in `<main>'

**Reply by Zeppelin-PV** [reply]
mit diesen .env Einträgen läuft der mqtt-collector

MQTT_INTERVAL=5
MQTT_HOST=192.168.1.90
MQTT_PORT=1883
MQTT_USERNAME=Ernst
MQTT_PASSWORD=15103662
INFLUX_ORG=solectrus
INFLUX_HOST=influxdb
INFLUX_SCHEMA=http
INFLUX_PORT=8086
INFLUX_TOKEN=my-super-secret-admin-token

MAPPING_0_TOPIC=zigbee2mqtt/Gefrierschrank
MAPPING_0_MEASUREMENT=Gefrierschrank
MAPPING_0_FIELD=power
MAPPING_0_TYPE=float

aber Daten kommen noch keine an

das Topic wird vom MQTT Explorer so angegeben
zigbee2mqtt/Gefrierschrank
dann steht da alles in geschwefter Klammer, - muss dann das Field power auch in Klammern ???
{"child_lock":"UNLOCK","countdown":0,"current":0.26,"energy":42.66,"indicator_mode":"off/on","linkquality":39,"power":33,"power_outage_memory":"restore","state":"ON","update":{"installed_version":192,"latest_version":192,"state":"idle"},"update_available":null,"voltage":233}

root@raspberrypi:/home/pi/solectrus# docker compose logs mqtt-collector
mqtt-collector-1  | MQTT collector for SOLECTRUS, Version 0.4.1, built at 2024-11-09T06:42:39.089Z
mqtt-collector-1  | https://github.com/solectrus/mqtt-collector
mqtt-collector-1  | Copyright (c) 2023-2024 Georg Ledermann and contributors, released under the MIT License
mqtt-collector-1  | 
mqtt-collector-1  | Using Ruby 3.3.6 on platform aarch64-linux-musl
mqtt-collector-1  | Subscribing from MQTT broker at mqtt://192.168.1.90:1883
mqtt-collector-1  | Pushing to InfluxDB at http://influxdb:8086, bucket my-solectrus-bucket
mqtt-collector-1  | 
mqtt-collector-1  | Subscribing to 1 topics:
mqtt-collector-1  | - zigbee2mqtt/ => Gefrierschrank:{"power"} (float)
mqtt-collector-1  |

**Reply by ledermann** [reply]
Es ist offenbar so, dass über dieses Topic JSON-Daten ankommen und du den Verbrauchswert (ich vermute, die Zahl bei "power") daraus extrahieren musst. In der Doku steht das hier: https://docs.solectrus.de/referenz/mqtt-collector/topics#mapping_x_json_key-optional

Also so:

```env
MAPPING_0_JSON_KEY=power
```

Außerdem muss natürlich diese neue Variable im `docker-compose.yml` unter `environment` des MQTT-Collectors ergänzt werden.

**Reply by Zeppelin-PV** [reply]
hallo Georg, darf ich Dich nochmals bemühen ...

mit einem, so denke ich, "außergewöhnlichen Konstrukt komme ich bis hierher - aber nicht weiter
in der .env ist der komplette string

{"child_lock":"UNLOCK","countdown":0,"current":0.26,"energy":42.66,"indicator_mode":"off/on","linkquality":39,"power":33,"power_outage_memory":"restore","state":"ON","update":{"installed_version":192,"latest_version":192,"state":"idle"},"update_available":null,"voltage":233}

im Log sieht man, dass der String ausgelesen wird, mit Werten, aber irgendwie nicht ins "Field" interpretiert

![grafik](https://github.com/user-attachments/assets/2b5f8477-cb51-4c04-bf82-072152512fba)
![grafik](https://github.com/user-attachments/assets/6932e992-683b-4b56-8fbd-3ba2e901edef)
![grafik](https://github.com/user-attachments/assets/cc6a6647-4982-4bb4-a0e6-8104490ddb40)
![grafik](https://github.com/user-attachments/assets/c0a589b4-c34f-482d-88d9-f0ac1aa52184)

**Reply by ledermann** [reply]
Eigentlich bist du bereits sehr weit gekommen. Du hast eine funktionierenden Verbindung zum MQTT-Broker und hast korrekt ein Topic abonniert. Dein MQTT-Broker liefert für das angegebene Topic einen JSON-String zurück, der die Daten enthält. 

Somit ist eigentlich alles gut, die Probleme sind lösbar, es geht nur noch um die Extraktion des Messwertes.

Um aus dem JSON das Attribut "power" zu extrahieren (alles andere aus dem JSON wird nicht benötigt), brauchst du keinen JSONPath, du kannst das einfach über `JSON_KEY` erledigen, also wie bereits erwähnt:

```env
MAPPING_0_JSON_KEY=power
```

JSONPath ist nur für komplexere Strukturen nötig (verschachtelt oder mit Arrays), das zu deinem Topic gelieferte JSON ist aber nicht komplex. Wenn du unbedingt JSONPath verwenden möchtest, kannst du das auch tun, dann aber wie folgt:

```env
MAPPING_0_JSON_PATH=$.power
```

Du hast stattdessen einen ganzen JSON-String in der `.env` aufgeführt, das ist völlig verkehrt. Der JSON-String wird ja vom MQTT-Broker geliefert und ändert sich ständig. Die Variablen `JSON_KEY` oder `JSON_PATH` sind dazu da, aus einem JSON-String jedes Mal ein Attribut zu extrahieren. 

Ein anderer Fehler ist die Angabe von `MAPPING_0_FIELD="power":` - der abschließende Doppelpunkt ist falsch und muss entfernt werden, die Anführungszeichen sind unnötig. Die korrekte Angabe ist:

```env
MAPPING_0_FIELD=power
```

Insgesamt sollte das Mapping in der `.env` so aussehen:

```env
# Abonniere dieses Topic vom MQTT-Broker, d.h. empfange Messwerte
MAPPING_0_TOPIC=zigbee2mqtt/Gefrierschrank

# Wenn ein JSON-String ankommt, dann extrahiere daraus den Wert für das Attribut "power"
MAPPING_0_JSON_KEY=power

# Schreibe das Resultat in die InfluxDB ins Measurement "Gefrierschrank" und dort ins Field "power" (als Dezimalzahl)
MAPPING_0_MEASUREMENT=Gefrierschrank
MAPPING_0_FIELD=power
MAPPING_0_TYPE=float
```

Diese vier Variablen müssen auch in der `docker-compose.yml` aufgeführt werden, aber das ist vermutlich bereits so.

Noch eine Anmerkung: Lass die Screenshots weg, insbesondere die von InfluxDB und SOLECTRUS sind unnötig, wenn im Log des MQTT-Collectors bereits erkennbar ist, dass etwas nicht funktioniert. Diese Zeile hier nämlich besagt, dass nur ein 0-Wert geschrieben wurde, obwohl im JSON ein Wert von 32 steht:

```
=> Gefrierschrank:power = 0.0
```

Was du erreichen musst, wäre eine solche Ausgabe:

```
# Message from 2024-12-09 12:34:56 +01:00
  topic = zigbee2mqtt/Gefrierschrank
  message = {"child_lock":"UNLOCK","countdown":0,"current":0.26,"energy":42.66,"indicator_mode":"off/on","linkquality":39,"power":33,"power_outage_memory":"restore","state":"ON","update":{"installed_version":192,"latest_version":192,"state":"idle"},"update_available":null,"voltage":233}
  => Gefrierschrank:power = 33.0
```

**Reply by Zeppelin-PV** [reply]
... ich bin sprachlos - Georg 
ich habe alle .env Varianten die ich probiert habe in einem Worddokument zwischengespeichert
war mit einer knapp dabei, aber eben nur knapp
nimmt man den richtigen Legostein, wird ein Haus draus
herzlichen Dank, an dieser Stelle

**Reply by rolkrz** [reply]
Hallo zusammen!
Ich bin gerade über diesen Thread gestoßen, weil ich ebenfalls Schwierigkeiten habe, eine über MQTT empfangene Nachricht korrekt zu parsen. Ziel bei mir ist es, Energiedaten von Tasmota-Geräte hier zu importieren. Ich habe ebenfalls laut Doku gesehen, dass eigentlich nur der MAPPING_X_JSOY_KEY gesetzt werden muss und es sollte passen. Der importierte Wert blieb aber stets 0.0. Also habe ich mal das hier oben beschriebene Beispiel 1:1 hergenommen. Ich habe in der .env Datei folgendes Mapping:

```
MAPPING_0_TOPIC=zigbee2mqtt/Gefrierschrank
MAPPING_0_JSON_KEY=power
MAPPING_0_MEASUREMENT=Gefrierschrank
MAPPING_0_FIELD=power
MAPPING_0_TYPE=float
```

Schicke ich jetzt den JSON-String {"power": 33.0} an das Topic zigbee2mqtt/Gefrierschrank bekomme ich im Log des mqtt-collector Containers trotzdem nur die Ausgabe:

```
# Message from 2025-05-04 15:32:11 +0200
  topic = zigbee2mqtt/Gefrierschrank
  message = {"power": 33.0}
  => Gefrierschrank:power = 0.0
```

Schicke ich allerdings nur die 33.0, passt alles:

```
# Message from 2025-05-04 15:33:21 +0200
  topic = zigbee2mqtt/Gefrierschrank
  message = 33.0
  => Gefrierschrank:power = 33.0
```

Was mache ich den noch falsch, dass der JSON String nicht als solcher erkannt bzw. nicht geparst wird?

Viele Grüße, Roland

**Reply by ledermann** [reply]
Vermutlich ist `MAPPING_0_JSON_KEY` in deinem `compose.yaml` nicht im Abschnitt `environment` deklariert, d.h. die Variable kommt im Container gar nicht an.

Falls das nicht hilft, bitte **in einem neuen Thema** weitere Informationen liefern, insbesondere den Inhalt von `.env` und `compose.yaml`.

> [!IMPORTANT]  
> Bitte zukünftig keine alten Threads mit neuen Fragen versehen, sondern bei neuen Fragen immer auch ein neues Thema aufmachen.

**Reply by rolkrz** [reply]
Tatsache! Das hat noch gefehlt. Nun funktioniert es. Vielen Dank hier für!
